### PR TITLE
AWS SDK V3 migration

### PR DESCRIPTION
**Pull request by gmrabian**
### Description

Upgrade MCR to use AWS SDK v3

This PR was greatly assisted by the efforts of @benmartin-coforma in the [upgrade for CARTS](https://github.com/Enterprise-CMCS/macpro-mdct-carts/pull/139521)

Tips for reviewing:
- Start with everything other than app-api
- Once you get to app-api start with package.json, then the utils/ folder
- Almost every change to a handler is updating how it uses the utils and new types
- Almost every change to a test is updating mocks for the new sdk

Notes:
- s3 get is different https://docs.aws.amazon.com/AmazonS3/latest/userguide/example_s3_GetObject_section.html
  - it now returns a streamed body, which we have to convert to a string and then parse. this is updated in the s3-lib with virtually no impact elsewhere (except mocks in tests)

References:
- AWS SDK v3 [docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/)

### Related ticket(s)

CMDCT-3074

---
### How to test

deployed url: https://d1jtat55rts1jz.cloudfront.net/

Click around and make sure everything is ok!

Our cypress tests cover everything that was changed here, so you can just validate whatever use cases you like.

### Important updates

yarn everything

---
### Author checklist

- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
---

### Feedback

I'm interested in feedback on better ways to handle the test mocking. They resulted in a lot of duplication, though I don't want to mock too globally for fear of having false positives.

### CODE REVIEW

**Review by karla-vm — DISMISSED** [review]
This looks good to me!

**Review by benmartin-coforma — APPROVED** [review]
This was a ton of work, well done.

File: services/app-api/utils/dynamo/dynamodb-lib.test.ts [hunk 1]
```diff
@@ -1,126 +1,84 @@
...
-      createDbClient();
-      expect(DynamoDB.DocumentClient).toHaveBeenCalledWith({
-        region: "us-east-1",
-      });
-    });
+    expect(result).toHaveLength(2);
+    expect(result[0]).toBe(mockItem1);
+    expect(result[1]).toBe(mockItem2);
+    expect(extraCall).not.toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (gmrabian) [reply services/app-api/utils/dynamo/dynamodb-lib.test.ts:53]: it's based entirely on the [`scanAll` test](https://github.com/Enterprise-CMCS/macpro-mdct-carts/blob/main/services/app-api/libs/tests/dynamodb-lib.test.ts#L68-L88) you wrote for CARTS 😆

File: services/app-api/utils/s3/s3-lib.test.ts [hunk 2]
```diff
@@ -1,52 +1,90 @@
...
+    s3ClientMock.reset();
+  });
+
+  test("Can get object", async () => {
+    // ignore type error on body as the mocked body is acceptable and matches use case
+    s3ClientMock.on(GetObjectCommand).resolves({
+      // @ts-ignore
+      Body: {
+        transformToString: () => Promise.resolve(`{"json":"blob"}`),
+      },
```
[the remark this answers is not in the document]
Comment (gmrabian) [reply services/app-api/utils/s3/s3-lib.test.ts:36]: I think I like `as GetObjectCommandOutput` so I'll go with that. Thanks!

File: services/app-api/utils/dynamo/dynamodb-lib.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (benmartin-coforma) [reply services/app-api/utils/dynamo/dynamodb-lib.test.ts:53]: 🤦  well i do like it

**Review by braxex — APPROVED** [review]

**Pull request merged**